content/_data/changelogs/_STYLEGUIDE.adoc

```diff
@@ -0,0 +1,56 @@
+= Jenkins Changelog Styleguide
+
+The Changelog describes notable changes in Jenkins.
+Not all changes are included, but all changes impacting users or developers are.
+
+== Types of changes
+
+The changelog supports five different types of changes:
+
+* *security* refers specifically to security fixes.
+  These changes are typically just references to a security advisory.
+* *major bug* are major bug fixes. *bug* are all other bug fixes.
+* *major rfe* are major new features and improvements. *rfe* are all other features and improvements.
+
+Library upgrades are generally assigned the label matching the motivation for the upgrade, which should be reflected in the label:
+_Upgraded foo from 1.2 to 1.4 to fix problems related to widget frobnication._
+
+When in doubt about the nature of a change, a change should be marked as an RFE and not a bug fix.
+
+Use the *major* label for changes that are very important, or have a high risk of introducing regressions (which often corresponds to the number of lines changed).
+
+== Labels
+
+* Use present tense to describe the specific change. "Fix a problem", rather than "Fixed a problem".
+  Descriptions of previous (now obsolete) behavior can deviate from this rule.
+* Use the "Developer:" prefix for any changes expected to only matter to developers of core or plugins.
+* Use the "Internal:" prefix for any notable internal changes with no expected impact outside core.
+* When upgrading a library, include both the former and current version number.
+* Use complete sentences as label, and end them with a period.
+** Changelog entries related to security fixes are an exception to this, as they only state _Security fixes_ or _Important security fixes_.
+   Use the latter label in case any fixes are considered high severity.
+
+== Messages
+
+Up to one highlighted message can be assigned to a Jenkins release.
+Reserve these messages for important announcements that go beyond regular changes.
+Examples include new required minimum Java versions, introduction of a system like JEP-200 with expected widespread impact, or when a release is very broken (e.g. imprope
+
+== Order of entries (regular)
+
+We generally order entries in descending order of importance.
+
+* Group entries by type ('major rfe' and 'rfe' are different types, the same applies to bugs).
+  An exception to this are entries that are only relevant to developers, or are considered 'internal' (no expected user impact).
+  Those are always listed last.
+* Put 'major' changes before non-major changes.
+* Always put 'security' entries first, if any.
+* After any 'security' entries, put the most important non-security change in the release.
+  This is typically a 'major rfe' or 'major bug', if any.
+* If in doubt, alternate *rfe* and *bug* entries to use the contrasting color to visually separate groups of entries.
+
```

Comment:
Add: "Put 'internal' and 'developer' changes in the bottom so that users firstly see changelogs relevant to them"?

Reply:
Yes, I've struggled with these. Right now it's part of the first item. Just noting this again would add redundancy.